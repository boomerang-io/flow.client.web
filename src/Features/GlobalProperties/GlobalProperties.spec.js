import React from "react";
import GlobalPropertiesContainer from "./index";
import { screen, fireEvent } from "@testing-library/react";
import { startApiServer } from "ApiServer";

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
});

describe("GlobalPropertiesContainer --- Snapshot Test", () => {
  it("Capturing Snapshot of GlobalPropertiesContainer", async () => {
    const { baseElement } = global.rtlQueryRender(<GlobalPropertiesContainer />);
    await screen.findByText(/Set global parameters that are available for all Workflows/i);

    expect(baseElement).toMatchSnapshot();
  });
});

describe("GlobalPropertiesContainer --- RTL", () => {
  it("rtlQueryRender the table and search correctly", async () => {
    global.rtlQueryRender(<GlobalPropertiesContainer />);
    await screen.findByText(/Set global parameters that are available for all Workflows/i);
    expect(screen.queryAllByText(/Test parameter/i).length).toBe(1);
    const searchProperty = screen.getByPlaceholderText(/Search/i);

    fireEvent.change(searchProperty, { target: { value: "2" } });

    expect(screen.queryAllByText(/Test parameter/i).length).toBe(0);

    fireEvent.change(searchProperty, { target: { value: "Test" } });

    expect(screen.queryAllByText(/Test parameter/i).length).toBe(1);
  });

  it("Opens create parameter modal", async () => {
    global.rtlQueryRender(<GlobalPropertiesContainer />);
    await screen.findByText(/Set global parameters that are available for all Workflows/i);

    expect(screen.queryByText(/CREATE parameter/)).not.toBeInTheDocument();

    const modalTrigger = screen.getByTestId("create-global-parameter-button");

    fireEvent.click(modalTrigger);

    expect(screen.getByText(/Cancel/)).toBeInTheDocument();
  });

  it("Opens edit parameter modal", async () => {
    global.rtlQueryRender(<GlobalPropertiesContainer />);
    await screen.findByText(/Set global parameters that are available for all Workflows/i);
    expect(screen.queryByText(/EDIT TEST RTL/i)).not.toBeInTheDocument();

    fireEvent.mouseOver(screen.getAllByTestId("configuration-parameter-table-row")[0]);

    const actionsMenu = screen.getAllByTestId("configuration-parameter-table-overflow-menu")[0];

    fireEvent.click(actionsMenu);

    const modalTrigger = await screen.findByText(/Edit/i);

    fireEvent.click(modalTrigger);

    expect(screen.getByText(/Edit Test parameter/i)).toBeInTheDocument();
  });

  it("Opens delete parameter modal", async () => {
    global.rtlQueryRender(<GlobalPropertiesContainer />);
    await screen.findByText(/Set global parameters that are available for all Workflows/i);
    expect(screen.queryByText(/DELETE TEST RTL/i)).not.toBeInTheDocument();

    fireEvent.mouseOver(screen.getAllByTestId("configuration-parameter-table-row")[0]);

    const actionsMenu = screen.getAllByTestId("configuration-parameter-table-overflow-menu")[0];

    fireEvent.click(actionsMenu);

    const modalTrigger = await screen.findByText(/Delete/i);

    fireEvent.click(modalTrigger);

    expect(screen.getByText(/Delete Test parameter?/i)).toBeInTheDocument();
  });
});
