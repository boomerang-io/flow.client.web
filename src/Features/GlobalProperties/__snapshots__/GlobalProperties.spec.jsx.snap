// Vitest Snapshot v1

exports[`GlobalPropertiesContainer --- Snapshot Test > Capturing Snapshot of GlobalPropertiesContainer 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_95546d"
    >
      <header
        class="cds--bmrg-feature-header _header_9e7615"
      >
        <section
          class="cds--bmrg-feature-header__content"
        >
          <div
            class="cds--bmrg-feature-header__header"
          >
            <h1
              class="cds--bmrg-feature-header-title _headerTitle_9e7615"
            >
              Parameters
            </h1>
            <p
              class="cds--bmrg-feature-header-title"
            >
              Set global parameters that are available for all Workflows
            </p>
          </div>
        </section>
      </header>
      <div
        class="_tableContainer_9e7615"
      >
        <div
          class="_tableHeader_9e7615"
        >
          <div
            aria-labelledby="parameters-table-search-search"
            class="cds--search cds--search--md _search_9e7615"
            role="search"
          >
            <div
              class="cds--search-magnifier"
            >
              <svg
                aria-hidden="true"
                class="cds--search-magnifier-icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                />
              </svg>
            </div>
            <label
              class="cds--label"
              for="parameters-table-search"
              id="parameters-table-search-search"
            >
              Search
            </label>
            <input
              autocomplete="off"
              class="cds--search-input"
              id="parameters-table-search"
              placeholder="Search"
              role="searchbox"
              type="text"
              value=""
            />
            <button
              aria-label="Clear search input"
              class="cds--search-close cds--search-close--hidden"
              title="Clear search input"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                />
              </svg>
            </button>
          </div>
          <div
            class="_actions_9e7615"
          >
            <button
              class="cds--btn cds--btn--md cds--btn--primary"
              data-testid="create-global-parameter-button"
              style="width: 12rem;"
              tabindex="0"
              type="button"
            >
              Create Parameter
              <svg
                aria-hidden="true"
                aria-label="Create Parameter"
                class="cds--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="cds--data-table-container"
        >
          <div
            class="cds--data-table-content"
          >
            <table
              class="cds--data-table cds--data-table--sort"
            >
              <thead>
                <tr
                  class="_tableHeadRow_9e7615"
                >
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_9e7615 _label_9e7615"
                    id="label"
                    scope="col"
                  >
                    <div
                      id="table-sort-3"
                      style="display: none;"
                    >
                      Click to sort rows by Label header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-3"
                      class="_tableHeadHeader_9e7615 _label_9e7615 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Label
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_9e7615 _key_9e7615"
                    id="key"
                    scope="col"
                  >
                    <div
                      id="table-sort-4"
                      style="display: none;"
                    >
                      Click to sort rows by Key header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-4"
                      class="_tableHeadHeader_9e7615 _key_9e7615 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Key
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_9e7615 _description_9e7615"
                    id="description"
                    scope="col"
                  >
                    <div
                      id="table-sort-5"
                      style="display: none;"
                    >
                      Click to sort rows by Description header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-5"
                      class="_tableHeadHeader_9e7615 _description_9e7615 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Description
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_9e7615 _value_9e7615"
                    id="value"
                    scope="col"
                  >
                    <div
                      id="table-sort-6"
                      style="display: none;"
                    >
                      Click to sort rows by Value header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-6"
                      class="_tableHeadHeader_9e7615 _value_9e7615 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Value
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="_tableHeadHeader_9e7615 _secured_9e7615"
                    id="secured"
                    scope="col"
                  >
                    <div
                      id="table-sort-7"
                      style="display: none;"
                    >
                      Click to sort rows by Secured header in ascending order
                    </div>
                    <button
                      aria-describedby="table-sort-7"
                      class="_tableHeadHeader_9e7615 _secured_9e7615 cds--table-sort"
                      type="button"
                    >
                      <span
                        class="cds--table-sort__flex"
                      >
                        <div
                          class="cds--table-header-label"
                        >
                          Secured
                        </div>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="cds--table-sort__icon-unsorted"
                          fill="currentColor"
                          focusable="false"
                          height="20"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 32 32"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    class="_tableHeadHeader_9e7615 _actions_9e7615"
                    id="actions"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                aria-live="polite"
                class="_tableBody_9e7615"
              >
                <tr
                  data-testid="configuration-parameter-table-row"
                >
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        Test parameter
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        test.key
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        adding a test description
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        Test value
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <svg
                        alt="unsecured"
                        aria-hidden="true"
                        class="_tableSecured_9e7615 _unsecured_9e7615"
                        fill="currentColor"
                        focusable="false"
                        height="32"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                        />
                      </svg>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Overflow menu"
                        class="cds--overflow-menu cds--overflow-menu--md"
                        data-testid="configuration-parameter-table-overflow-menu"
                        type="button"
                      >
                        <svg
                          aria-label="Overflow menu icon"
                          class="cds--overflow-menu__icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          role="img"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="16"
                            cy="8"
                            r="2"
                          />
                          <circle
                            cx="16"
                            cy="16"
                            r="2"
                          />
                          <circle
                            cx="16"
                            cy="24"
                            r="2"
                          />
                        </svg>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  data-testid="configuration-parameter-table-row"
                >
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        ben.label
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        ben.key
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        ---
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <p
                        class="_tableTextarea_9e7615"
                      >
                        ---
                      </p>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <svg
                        alt="secured"
                        aria-hidden="true"
                        class="_tableSecured_9e7615 _secured_9e7615"
                        fill="currentColor"
                        focusable="false"
                        height="32"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 24L4 15 5.414 13.586 13 21.171 26.586 7.586 28 9 13 24z"
                        />
                      </svg>
                    </div>
                  </td>
                  <td>
                    <div
                      class="_tableCell_9e7615"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Overflow menu"
                        class="cds--overflow-menu cds--overflow-menu--md"
                        data-testid="configuration-parameter-table-overflow-menu"
                        type="button"
                      >
                        <svg
                          aria-label="Overflow menu icon"
                          class="cds--overflow-menu__icon"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          role="img"
                          viewBox="0 0 32 32"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="16"
                            cy="8"
                            r="2"
                          />
                          <circle
                            cx="16"
                            cy="16"
                            r="2"
                          />
                          <circle
                            cx="16"
                            cy="24"
                            r="2"
                          />
                        </svg>
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="cds--pagination cds--pagination--md"
        >
          <div
            class="cds--pagination__left"
          >
            <label
              class="cds--pagination__text"
              for="cds-pagination-select-id-14"
              id="cds-pagination-select-id-14-count-label"
            >
              Items per page:
            </label>
            <div
              class="cds--form-item cds--select__item-count"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-14"
                    >
                      <option
                        class="cds--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="cds--select-option"
                        value="25"
                      >
                        25
                      </option>
                      <option
                        class="cds--select-option"
                        value="50"
                      >
                        50
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text cds--pagination__items-count"
            >
              1–2 of 2 items
            </span>
          </div>
          <div
            class="cds--pagination__right"
          >
            <div
              class="cds--form-item cds--select__page-number"
            >
              <div
                class="cds--select cds--select--inline"
              >
                <label
                  class="cds--label cds--visually-hidden"
                  for="cds-pagination-select-id-14-right"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="cds--select-input--inline__wrapper"
                >
                  <div
                    class="cds--select-input__wrapper"
                  >
                    <select
                      class="cds--select-input"
                      id="cds-pagination-select-id-14-right"
                    >
                      <option
                        class="cds--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="cds--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="cds--pagination__text"
            >
              of 1 page
            </span>
            <div
              class="cds--pagination__control-buttons"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-11"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--backward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 24L10 16 20 8z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-11"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Previous page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top-right cds--tooltip cds--icon-tooltip"
              >
                <button
                  aria-labelledby="tooltip-13"
                  class="cds--btn--icon-only cds--pagination__button cds--pagination__button--forward cds--pagination__button--no-index cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
                  disabled=""
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 8L22 16 12 24z"
                    />
                  </svg>
                </button>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-13"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Next page
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
</body>
`;
